mvc/views/dependency-injection.rst

```diff
@@ -1,93 +1,79 @@
 Injecting a Service Into a View
 ===============================
 
-ASP.NET MVC 6 now supports injection into a view from a class. For this example, we'll create a simple class that exposes the total *todo* count, completed count and average priority. 
-
-1. Examine the *Services\\StatisticsService.cs* class.
-
-  .. code-block:: c#
-    :linenos:
-    
-    using System.Linq;
-    using System.Threading.Tasks;
-    using TodoList.Models;
-
-    namespace TodoList.Services
-    {
-      public class StatisticsService
-      {
-        private readonly ApplicationDbContext db;
-
-        public StatisticsService(ApplicationDbContext context)
-        {
-          db = context;
-        }
-
-        public async Task<int> GetCount()
-        {
-          return await Task.FromResult(db.TodoItems.Count());
-        }
-
-        public async Task<int> GetCompletedCount()
-        {
-          return await Task.FromResult(
-            db.TodoItems.Count(x => x.IsDone == true));
-        }
-
-        public async Task<double> GetAveragePriority()
-        {
-          if (db.TodoItems.Count() == 0)
-          {
-            return 0.0;
-          }
-
-          return await Task.FromResult(
-            db.TodoItems.Average(x =>x.Priority));
-        }
-      }
-    }
-
-2. Update the *Index* view to inject the *todo* statistical data. Add the ``inject`` statement to the top of the file:
-
-.. code-block:: html 
+By `Steve Smith`_, `Rick Anderson`_
+
+ASP.NET MVC 6 supports `dependency injection <https://docs.asp.net/en/latest/fundamentals/dependency-injection.html>`_ into views. This can be useful for view-specific services, such as localization or data required only for populating view elements.
+
+.. TODO: Add localization sample in RC2 timeframe.
+
+`View sample files <https://github.com/aspnet/Docs/tree/1.0.0-rc1/mvc/views/dependency-injection/sample>`_
+
+.. warning:: When applying this functionality, ensure you maintain `separation of concerns <http://deviq.com/separation-of-concerns>`_ between your controllers and views. Most of the data your views  display should be passed in from the controller. This technique is recommended only for view-specific data.
+
+A Simple Example
+----------------
+With the above warning in mind, you can inject a service into a view using the ``@inject`` directive, as shown below:
```

Comment:
The code below (from the http://docs.asp.net/projects/mvc/en/latest/views/view-components.html sample), shows injecting a service into a view using the @inject directive: